modules/internal/src/main/scala/higherkindness/mu/rpc/internal/serviceImpl.scala

```diff
@@ -338,47 +386,229 @@ object serviceImpl {
             case _ =>
               sys.error(
-                s"Unable to define a handler for the streaming type $streamingType and $streamingImpl for the method $methodName in the service $serviceName")
+                s"Unable to define a handler for the streaming type $streamingType and $prevalentStreamingTarget for the method $name in the service $serviceName")
           }
           q"($methodDescriptorName, $handler)"
         }
       }
+
+      case class HttpOperation(operation: Operation) {
+
+        import operation._
+
+        val uri = name.toString
+
+        val method: TermName = request match {
+          case _: EmptyTpe => TermName("GET")
+          case _           => TermName("POST")
+        }
+
+        val executionClient: Tree = response match {
+          case MonixObservableTpe(_, _) =>
+            q"Observable.fromReactivePublisher(client.stream(request).flatMap(_.asStream[${response.safeInner}]).toUnicastPublisher)"
+          case Fs2StreamTpe(_, _) =>
+            q"client.stream(request).flatMap(_.asStream[${response.safeInner}])"
+          case _ =>
+            q"client.expectOr[${response.safeInner}](request)(handleResponseError)"
+        }
+
+        val requestTypology: Tree = request match {
+          case _: UnaryTpe =>
+            q"val request = Request[F](Method.$method, uri / ${uri.replace("\"", "")}).withEntity(req.asJson)"
+          case _: Fs2StreamTpe =>
+            q"val request = Request[F](Method.$method, uri / ${uri.replace("\"", "")}).withEntity(req.map(_.asJson))"
+          case _: MonixObservableTpe =>
+            q"val request = Request[F](Method.$method, uri / ${uri.replace("\"", "")}).withEntity(req.toReactivePublisher.toStream.map(_.asJson))"
+          case _ =>
+            q"val request = Request[F](Method.$method, uri / ${uri.replace("\"", "")})"
+        }
+
+        val responseEncoder =
+          q"""implicit val responseDecoder: EntityDecoder[F, ${response.safeInner}] = jsonOf[F, ${response.safeInner}]"""
+
+        def toRequestTree: Tree = request match {
+          case _: EmptyTpe =>
+            q"""def $name(implicit client: _root_.org.http4s.client.Client[F]): ${response.getTpe} = {
+		                  $responseEncoder
+		                  $requestTypology
+		                  $executionClient
+		                 }"""
+          case _ =>
+            q"""def $name(req: ${request.getTpe})(implicit client: _root_.org.http4s.client.Client[F]): ${response.getTpe} = {
+		                  $responseEncoder
+		                  $requestTypology
+		                  $executionClient
+		                 }"""
+        }
+
+        val routeTypology: Tree = (request, response) match {
+          case (_: Fs2StreamTpe, _: UnaryTpe) =>
+            q"""val requests = msg.asStream[${operation.request.safeInner}]
+              Ok(handler.${operation.name}(requests).map(_.asJson))"""
```

Comment:
In all the macros stuff we need FQN

Reply:
Done at https://github.com/higherkindness/mu/pull/552/commits/9b0a8f3c29fe1b00b30c08db8cb695f5bbdaf252